test/e2e/storage/pvc_protection.go

```diff
@@ -45,14 +49,30 @@ var _ = utils.SIGDescribe("PVC Protection", func() {
 		nameSpace = f.Namespace.Name
 		framework.ExpectNoError(framework.WaitForAllNodesSchedulable(client, framework.TestContext.NodeSchedulableTimeout))
 
+		defaultSC := getDefaultStorageClass(client)
+		sc := defaultSC
+		scToDelete = nil
+		// Class having WaitForFirstConsumer means the PVC won't become Bound: create and use a class with Immediate
+		if defaultSC.VolumeBindingMode != nil && *defaultSC.VolumeBindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
+			By("Creating a StorageClass with VolumeBindingMode Immediate")
+			sc = defaultSC.DeepCopy()
+			sc.ObjectMeta = metav1.ObjectMeta{
+				GenerateName: defaultSC.Name,
+			}
+			immediateMode := storagev1.VolumeBindingImmediate
+			sc.VolumeBindingMode = &immediateMode
+			sc, err = client.StorageV1().StorageClasses().Create(sc)
+			Expect(err).NotTo(HaveOccurred(), "Error creating a StorageClass with VolumeBindingMode Immediate")
+			scToDelete = sc
+		}
+
```

Comment:
Can we not make this test to create its own SC with bindingMode set to immediate?

Reply:
yeah. idk it saves some time/cycles if we use the default sc where possible, 'cause then we don't need to wait for the pv to be deleted etc.

Reply:
the only thing that changes is the if statement. because I need to clone the default sc anyway

Reply:
I dunno but the if statement bothers me. :-) 
We don't need to clone the default SC. We already have a `testStorageClass` variable which can be used to get a SC via https://github.com/kubernetes/kubernetes/blob/master/test/e2e/storage/volume_provisioning.go#L1031. We can also specify bindingMode in `testStorageClass` variable.

Reply:
I don't want to use newStorageClass because then I need to restrict the tests to the supported cloud providers (gce aws openstack vsphere azure). this test should not have any dependency on the provider, I should be allowed to run it locally or with whatever StorageClass happens to be default :/